File: QRStockMateSL/QRStockMate.AplicationCore/Entities/TransactionHistory.cs
using Swashbuckle.AspNetCore.Annotations;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QRStockMate.AplicationCore.Entities
{
     public class TransactionHistory
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public DateTime Created { get; set; }
        public OperationHistory Operation { get; set; }

    }

	[SwaggerSchema("0: Add, 1: Post, 2: Put, 3: Delete")]
	public enum OperationHistory { 
        Add,
        Post,
        Put,
        Delete
    }
}

File: QRStockMateWeb/src/app/interfaces/transaction-history.ts
import { User } from "./user";

export interface TransactionHistory {
  id: number;
  name: string;
  code: string;
  description: string;
  created: Date;
  operation: OperationHistory | Number;
}

export enum OperationHistory {
  Add = "Add",
  Post = "Post",
  Put = "Put",
  Delete = "Delete"
}
// Función para obtener el índice de un valor en el enum
export function getIndexFromOperation(operation: OperationHistory): Number {
  const keys = Object.keys(OperationHistory).filter(key => OperationHistory[key as keyof typeof OperationHistory] === operation);
  if (keys.length === 1) {
    return parseInt(keys[0]);
  }
  return 0;
}

export function me(): User | null{
  var stringU = sessionStorage.getItem('me')
  var me:User
  if (stringU) {
    me  = JSON.parse(stringU)
    return me;
  }

  return null;
}

export function token(): string{
  var stringT = sessionStorage.getItem('token')
  var token:string
  if (stringT) {
    token  = JSON.parse(stringT)
    return token;
  }

  return "";
}
